juju/juju
# 2.6 capped logs

This work changes the logs collections to be capped collections. This will stop the logs database growing without bound when we are writing logs faster than we can delete them.

The capping and sizing thereof is done when the database is opened.

This means we can remove the log pruner. We need to keep the controller config values for max-logs-age and max-logs-size as these values are required for earlier 2.6 releases and the later 2.6 client should still be able to bootstrap earlier 2.6 versions, and without these values, the apiserver panics.
 
## QA steps
```
* bootstrap a controller with debug logging
* deploy some things into the default model
* go into the mongo shell and check the collStats of the logs databases
* change the model-logs-size controller config value
* restart the controllers
* go into the mongo shell and check the collStats of the logs databases
```

```
* bootstrap a 2.6.3 controller with debug logging
* deploy some things into the default model
* upgrade the controller
* go into the mongo shell and check the collStats of the logs databases
```

## Documentation changes

Will need to update the controller config values.
